packages/compass-components/src/components/leafy-green/combobox/combobox-option.tsx

```diff
@@ -0,0 +1,196 @@
+import { css, cx } from '@leafygreen-ui/emotion';
+import React, { useCallback, useContext, useMemo } from 'react';
+import { uiColors } from '@leafygreen-ui/palette';
+import { isComponentType } from '@leafygreen-ui/lib';
+import { useForwardedRef, useIdAllocator } from '@leafygreen-ui/hooks';
+import Checkbox from '@leafygreen-ui/checkbox';
+import Icon, { isComponentGlyph } from '@leafygreen-ui/icon';
+import type { InternalComboboxOptionProps } from './combobox-types';
+import { ComboboxContext } from './combobox-context';
+import { wrapJSX } from './util';
+
+/**
+ * Styles
+ */
+const comboboxOptionStyle = () => css`
+  position: relative;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  list-style: none;
+  color: inherit;
+  cursor: pointer;
+  overflow: hidden;
+  font-size: var(--lg-combobox-item-font-size);
+  line-height: var(--lg-combobox-item-line-height);
+  padding: var(--lg-combobox-item-padding-y) var(--lg-combobox-item-padding-x);
+
+  &:before {
+    content: '';
+    position: absolute;
+    left: 0;
+    width: 3px;
+    height: var(--lg-combobox-item-wedge-height);
+    background-color: transparent;
+    border-radius: 0 2px 2px 0;
+    transform: scaleY(0.3);
+    transition: 150ms ease-in-out;
+    transition-property: transform, background-color;
+  }
+
+  &:hover {
+    outline: none;
+    background-color: var(--lg-combobox-item-hover-color);
+  }
+
+  &[aria-selected='true'] {
+    outline: none;
+    background-color: var(--lg-combobox-item-active-color);
+
+    &:before {
+      background-color: var(--lg-combobox-item-wedge-color);
+      transform: scaleY(1);
+    }
+  }
+`;
+
+const flexSpan = css`
+  display: inline-flex;
+  gap: 8px;
+  justify-content: start;
+  align-items: inherit;
+`;
+
+const displayNameStyle = (isSelected: boolean) => css`
+  font-weight: ${isSelected ? 'bold' : 'normal'};
+`;
+
+/**
+ * Component
+ */
+const InternalComboboxOption = React.forwardRef<
+  HTMLLIElement,
+  InternalComboboxOptionProps
+>(
+  (
+    {
+      displayName,
+      glyph,
+      isSelected,
+      isFocused,
+      setSelected,
+      className,
+    }: InternalComboboxOptionProps,
+    forwardedRef
+  ) => {
+    const { multiselect, darkMode, withIcons, inputValue } =
+      useContext(ComboboxContext);
+    const optionTextId = useIdAllocator({ prefix: 'combobox-option-text' });
+    const optionRef = useForwardedRef(forwardedRef, null);
+
+    const handleOptionClick = useCallback(
+      (e: React.SyntheticEvent) => {
+        e.stopPropagation();
+        // If user clicked on the option, or the checkbox
+        // Ignore extra click events on the checkbox wrapper
+        if (
+          e.target === optionRef.current ||
+          (e.target as Element).tagName === 'INPUT' ||
+          (e.target as Element).tagName === 'SPAN' // TODO(alenakhineika): investigate this workaround later.
```

Comment:
Here for some reason, the event did not happen on the span name of the li element.

Reply:
Seems fine, would it be possible to provide a link to the original code in the comment?